### PR TITLE
Fix broken JEDI container builds after updating ESMF and MAPL to 8.6.1/2.46.2

### LINKED ISSUE #1152: JEDI CI container build broken since update to esmf@8.6.1/mapl@2.46.2
**Describe the bug**
The CI containers don't build anymore since the update to esmf@8.6.1/mapl@2.46.2. This is because of a hardcoded pflogger dependency that doesn't match with what the new mapl needs.

**To Reproduce**
Look at CI

**Expected behavior**
It builds

**System:**
CI container builds (any)

**Additional context**
@climbfuji is fixing this in #1150

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

Update `configs/containers/specs/jedi-ci.yaml:` remove harcoded versions for `gftl-shared` (and unrelated, but needed for one of the next PRs #1138, for `nlohmann-json-*` ).

### Testing

CI container builds (run via workflow dispatch):
- [x] clang-mpich - https://github.com/JCSDA/spack-stack/actions/runs/9602152092/job/26482348472
- [x] gcc-openmpi - https://github.com/JCSDA/spack-stack/actions/runs/9605347366/job/26492772806
- [x] intel-impi - https://github.com/JCSDA/spack-stack/actions/runs/9607351473/job/26498364779

### Applications affected

JCSDA CI applications

### Systems affected

JCSDA CI

### Dependencies

n/a

### Issue(s) addressed

Resolves #1152 

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA FYI we'll need to make changes along those lines for your packages.yaml-late-spring-cleanup PR

File: configs/containers/specs/jedi-ci.yaml [hunk 1]
```diff
@@ -2,10 +2,9 @@
   specs: [base-env@1.0.0, jedi-base-env@1.0.0, ewok-env@1.0.0, jedi-fv3-env@1.0.0,
     jedi-mpas-env@1.0.0, bacio@2.4.1, bison@3.8.2, bufr@12.0.1, ecbuild@3.7.2, eccodes@2.33.0, ecflow@5,
     eckit@1.24.5, ecmwf-atlas@0.36.0 +fckit +trans +tesselation +fftw, fiat@1.2.0, ectrans@1.2.0 +fftw,
-    eigen@3.4.0, fckit@0.11.0, fms@2023.04, g2@3.4.9, g2tmpl@1.10.2, gftl-shared@1.6.1,
+    eigen@3.4.0, fckit@0.11.0, fms@2023.04, g2@3.4.9, g2tmpl@1.10.2,
     gsibec@1.2.1, hdf@4.2.15, hdf5@1.14.3, ip@5.0.0, jasper@2.0.32, jedi-cmake@1.4.0,
-    libpng@1.6.37, nccmp@1.9.0.1, netcdf-c@4.9.2, netcdf-cxx4@4.3.1,
-    netcdf-fortran@4.6.1, nlohmann-json@3.10.5, nlohmann-json-schema-validator@2.1.0,
+    libpng@1.6.37, nccmp@1.9.0.1, netcdf-c@4.9.2, netcdf-cxx4@4.3.1, netcdf-fortran@4.6.1,
```
Comment (srherbener) [inline configs/containers/specs/jedi-ci.yaml:7]: Doesn't the removal of the `nlohmann-json` and `nlohmann-json-schema-validator` packages disable the validation of all the YAML configuration in JEDI? Or are these packages still getting picked up as dependencies in the build? Just want to make sure that all of the oops::Parameters::validateAndDeserialize() function calls are still validating.

Comment (climbfuji) [reply configs/containers/specs/jedi-ci.yaml:7]: nlohmann-json are dependencies of jedi-fv3-env, which is listed above. This just removes the hardcoded versions.

Comment (srherbener) [reply configs/containers/specs/jedi-ci.yaml:7]: Great! Thanks for the explanation @climbfuji

**Review by srherbener — APPROVED** [review]

**Pull request merged**